modin/core/storage_formats/pandas/parsers.py

```diff
@@ -784,28 +784,77 @@ class ParquetFileToRead(NamedTuple):
 class PandasParquetParser(PandasParser):
     @staticmethod
     def _read_row_group_chunk(
-        f, row_group_start, row_group_end, columns, engine, to_pandas_kwargs
+        f, row_group_start, row_group_end, columns, filters, engine, to_pandas_kwargs
     ):  # noqa: GL08
         if engine == "pyarrow":
-            from pyarrow.parquet import ParquetFile
+            if filters is not None:
+                import pyarrow.dataset as ds
+                from pyarrow.parquet import filters_to_expression
 
-            return (
-                ParquetFile(f)
-                .read_row_groups(
-                    range(
+                parquet_format = ds.ParquetFileFormat()
+                fragment = parquet_format.make_fragment(
+                    f,
+                    row_groups=range(
                         row_group_start,
                         row_group_end,
                     ),
+                )
+                dataset = ds.FileSystemDataset(
+                    [fragment],
+                    schema=fragment.physical_schema,
+                    format=parquet_format,
+                    filesystem=fragment.filesystem,
+                )
+
+                # This lower-level API doesn't have the ability to automatically handle pandas metadata
+                # The following code inspired by
+                # https://github.com/apache/arrow/blob/f44e28fa03a64ae5b3d9352d21aee2cc84f9af6c/python/pyarrow/parquet/core.py#L2619-L2628
+
+                # if use_pandas_metadata, we need to include index columns in the
+                # column selection, to be able to restore those in the pandas DataFrame
+                metadata = dataset.schema.metadata or {}
+
+                if metadata and b"pandas" in metadata and columns is not None:
+                    import json
+
+                    index_columns = json.loads(metadata[b"pandas"].decode("utf8"))[
+                        "index_columns"
+                    ]
+                    # RangeIndex can be represented as dict instead of column name
+                    index_columns = [
+                        col for col in index_columns if not isinstance(col, dict)
+                    ]
```

Comment:
I copied this comment from pyarrow, along with most of the code around it, as noted by the comment above.

https://github.com/apache/arrow/blob/f44e28fa03a64ae5b3d9352d21aee2cc84f9af6c/python/pyarrow/parquet/core.py#L2619-L2628

I believe what this means is that in the pandas metadata, the index columns can either be string column names, or a dictionary that describes a RangeIndex. The point of this code block is to find the **real columns** that need to be read to part of the pandas Index, so it skips the RangeIndex.

Also note there is test coverage of this.

Reply:
> in the pandas metadata, the index columns can either be string column names, or a dictionary that describes a RangeIndex. The point of this code block is to find the **real columns** that need to be read to part of the pandas Index, so it skips the RangeIndex.

This would be awesome to have as in-code comment for future us, please add 😺